### ISSUE TITLE
Data Leakage

### ISSUE THREAD

**Issue by nickhalmagyi** [avisoori-databricks/e2e_ml_demo#1]
Hi, I like this e2e taxi notebook, it helped me understand feature stores.

But I think there is some bad practice in terms of data leakage going on.
I found that the dataset has

first_ts: datetime.datetime(2016, 1, 1, 0, 4, 30)
last_ts: datetime.datetime(2016, 2, 29, 23, 51, 20)

however you compute aggregated features  which over this whole time period, before you perform a train/test split. This means that the training set in principle knows something about the test set, through the aggregated features in the feature store. This may or may not have a huge effect on the performance of the model but unless I am mistaken, this is data leakage and should be avoided.

It's potentially a big deal, because from my point of view, one has to create a feature store for every train/test split, if one is computing aggregated features. 

If one is not computing aggregated features, it would be better to compute them in a pipeline I think.

If I'm missing something, I'd be very happy to know :)
